src/libstd/process.rs

```diff
@@ -1443,7 +1443,7 @@ pub fn id() -> u32 {
 #[cfg_attr(not(test), lang = "termination")]
 #[unstable(feature = "termination_trait_lib", issue = "43301")]
 #[rustc_on_unimplemented =
-  "`main` can only return types like `()` that implement {Termination}, not `{Self}`"]
+  "`main` can only return types that implement {Termination}, not `{Self}`"]
```

Comment:
You could change this to:

```
#[rustc_on_unimplemented(
  message="`main` has invalid return type `{Self}`",
  label="`main` can only return types that implement {Termination}")]
```

So that the output is

```
error[E0277]: `main` has invalid return type `i32`
  --> src/bin/termination-trait-main-i32.rs:11:14
   |
11 | fn main() -> i32 {
   |              ^^^ `main` can only return types that implement `std::process::Termination`
   |
   = help: consider using `()`, or a `Result`
```

Reply:
That works. I wasn't sure how important going back to the original error code was.